### PR TITLE
Distribution model

### PR DESCRIPTION
**Pull request by drondin**
Calculate profitsharingRatio and buybackRate

### CODE REVIEW

**Comment by drondin** [comment]
I still need to add tests to DoHardWorkTest with different scenarios.

File: src/main/java/pro/belbix/ethparser/web3/harvest/parser/HardWorkParser.java [hunk 1]
```diff
@@ -192,7 +226,7 @@ private void parseRewardAddedEvents(Log ethLog, HardWorkDTO dto, boolean autoSta
       // AutoStake strategies have two RewardAdded events - first for PS and second for stake contract
       if (autoStake && dto.getFarmBuyback() != 0) {
         // in this case it is second reward for strategy
-        double fullReward = (reward * dto.getFarmPrice()) / 0.7; // full reward
+        double fullReward = (reward * dto.getFarmPrice()) / (1-dto.getProfitSharingRate()); // full reward
```
Comment (belbix) [inline src/main/java/pro/belbix/ethparser/web3/harvest/parser/HardWorkParser.java:229]: check on null and use default value 0.7

File: src/main/java/pro/belbix/ethparser/web3/harvest/parser/HardWorkParser.java [hunk 2]
```diff
@@ -201,7 +235,7 @@ private void parseRewardAddedEvents(Log ethLog, HardWorkDTO dto, boolean autoSta
         // for non AutoStake strategy we will not have accurate data for strategy reward
         // just calculate aprox value based on PS reward
         if (!autoStake) {
-          double fullReward = ((reward * dto.getFarmPrice()) / 0.3); // full reward
+          double fullReward = ((reward * dto.getFarmPrice()) / dto.getProfitSharingRate()); // full reward
```
Comment (belbix) [inline src/main/java/pro/belbix/ethparser/web3/harvest/parser/HardWorkParser.java:238]: default value

File: src/main/java/pro/belbix/ethparser/web3/harvest/parser/HardWorkParser.java [hunk 3]
```diff
@@ -161,6 +162,39 @@ private void parseRewards(HardWorkDTO dto, String txHash, String strategyHash) {
     fillFeeInfo(dto, txHash, tr);
   }
 
+  private void parseRates(HardWorkDTO dto, String strategyHash) {
+    double profitSharingDenominator =  functionsUtils.callIntByName("profitSharingDenominator", strategyHash, dto.getBlock()).orElse(BigInteger.ZERO).doubleValue();
```
Comment (belbix) [inline src/main/java/pro/belbix/ethparser/web3/harvest/parser/HardWorkParser.java:166]: I think better throw an exception when call failed

**Comment by drondin** [comment]
It is ready but after merging branch I get:

```
[ERROR] /home/user/workspace/ethparser/src/main/java/pro/belbix/ethparser/web3/layers/blocks/downloader/EthBlockDownloader.java:[60,7] cannot find symbol
[ERROR]   symbol:   variable log
[ERROR]   location: class pro.belbix.ethparser.web3.layers.blocks.downloader.EthBlockDownloader
[ERROR] /home/user/workspace/ethparser/src/main/java/pro/belbix/ethparser/web3/layers/blocks/downloader/EthBlockDownloader.java:[67,7] cannot find symbol
[ERROR]   symbol:   variable log
[ERROR]   location: class pro.belbix.ethparser.web3.layers.blocks.downloader.EthBlockDownloader
```

**Pull request merged**